frontend/catalyst/compilation_pipelines.py

```diff
@@ -634,10 +634,9 @@ def compute_jvp(self, primals, tangents):
 
         # Optimization: Do not compute Jacobians for arguments which do not participate in
         #               differentiation.
-        # TODO: replace with symbolic zero support
         argnums = []
         for idx, tangent in enumerate(tangents):
-            if qml.math.is_abstract(tangent, like="jax") or jnp.any(jnp.atleast_1d(tangent) != 0.0):
+            if not isinstance(tangent, jax.custom_derivatives.SymbolicZero):
```

Comment:
Will JAX ever pass in a materialized zero tangent (and do we care)? Or is it enough to make a decision based on JAX's behavior?

Reply:
Well with the option for symbolic zeros they shouldn't materialize it if they know the arg doesn't participate in differentiation. Now I think in theory that doesn't prevent the rare case of the tangent vector just happening to be *numerically* zero, but I don't know if we need to consider that.